src/components/create-safe/status/usePendingSafeCreation.ts

```diff
@@ -33,15 +54,7 @@ export const checkSafeCreationTx = async (
   const TIMEOUT_TIME = 6.5 * 60 * 1000 // 6.5 minutes
 
   try {
-    const txResponse = await provider.getTransaction(txHash)
-
-    if (!txResponse) {
-      return new Promise((resolve) => {
-        setTimeout(() => {
-          resolve(SafeCreationStatus.TIMEOUT)
-        }, TIMEOUT_TIME)
-      })
-    }
+    const txResponse = await pollTransaction(provider, txHash)
```

Comment:
Not an ideal solution to poll `getTransaction` imo since `waitForTransaction` also calls `getTransaction` internally. Alternatively, we could encode and cache the original tx info on the client side and reuse that to build the `replacement` object but it requires a bigger change.